### DISCUSSION TITLE
Feature request: drawing on surfaces

### DISCUSSION THREAD

**Discussion by geheiligt** [RomanArzumanyan/VALI#100] [Ideas]
Hi,
I actually doubt that this is the scope of this library, but it would be really nice to have gpu accelerated drawing functionality (boxes, points, lines, text...)  
Use case: Fast video transcoding with adding dynamic overlays (i.e. inference results)

Currently it is neccessary to dowload a planar rgb surface, annotate with i.e. Pillow, upload to a surface again.
VALI already supports decoding/encoding it would be efficient to also be able to do simple annotations directly in gpu ram.

Kindof like what these cuda projects are doing [cvcuda OSD](https://cvcuda.github.io/CV-CUDA/_python_api/_cvcuda_api/_op_osd.html) / [jetson_utils draw](https://github.com/dusty-nv/jetson-inference/blob/master/docs/aux-image.md#drawing-shapes)

**Comment by RomanArzumanyan** [comment]
Hi @geheiligt 

CV-CUDA does support DLPack (same as VALI) so I can't see any particular reason why it can't be used for everything OSD.
To me it looks like the lack of sample. BTW did you try CV-CUDA <> VALI interop ? Just asking, I never did, but definitely should.

**Comment by RomanArzumanyan** [comment]
Hi @geheiligt 

It turns out that VALI works just fine with `cvcuda`.
Please take a look at the sample: https://github.com/RomanArzumanyan/VALI/blob/main/samples/sample_segmentation_cvcuda_interop.ipynb

No re-build or re-install is required, DLPack memory sharing works already.
